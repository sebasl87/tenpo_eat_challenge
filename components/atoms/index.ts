export { default as RestaurantsLayout } from './RestaurantsLayout';
