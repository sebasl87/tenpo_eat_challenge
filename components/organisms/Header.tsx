import React from 'react'
import { Image, Text, View } from 'react-native';
import { styles } from '../../styles/styles';

function Header() {
    return (
        <>
            <View style={styles.header}>
                <Image
                    source={require("../../assets/imgs/user.png")}
                    style={styles.userImg}
                />
                <Image
                    source={require("../../assets/icons/search.png")}
                    style={styles.searchImg}
                />
            </View>
            <View style={styles.header}>
                <View style={{ flexDirection: "column" }}>
                    <Image
                        source={require("../../assets/imgs/logo.png")}
                        style={styles.logoImg}
                    />
                    <Text>DELIVER APP</Text>
                </View>
                <View style={{ flexDirection: "row", marginBottom: -16 }}>
                    <Image
                        source={require("../../assets/imgs/hand.png")}
                        style={styles.handImg}
                    />
                    <View style={{ top: "100%", height: 6 }}>

                        <Image
                            source={require("../../assets/icons/ellipse.png")}
                            style={styles.ellipseImg}
                        />
                    </View>
                </View>
            </View>
        </>
    );
}

export default Header