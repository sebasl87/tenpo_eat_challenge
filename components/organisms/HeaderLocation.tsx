import React from 'react'
import { Image, Text, TextInput, View } from 'react-native';
import { styles } from '../../styles/styles';

function Header() {
    return (
        <>
            <View style={styles.headerLocation}>
                <View style={{ marginBottom: 32, display: "flex", flexDirection: "row" }}>
                    <Image
                        source={require("../../assets/icons/location.png")}
                        style={{ height: 24, width: 24, marginRight: 6, paddingTop: 30 }}
                    />
                    <Text style={styles.fontTitle}>Agregar dirección de la entrega</Text>
                </View>
            </View>
            <TextInput placeholder='Escribe tu dirección' style={styles.inputLocation} />
        </>
    );
}

export default Header