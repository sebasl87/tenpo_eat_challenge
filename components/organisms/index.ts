export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
