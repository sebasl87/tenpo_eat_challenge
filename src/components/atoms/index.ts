import AddressLayout from './AddressLayout';
export { default as RestaurantsLayout } from './RestaurantsLayout';
export { default as CategoriesLayout } from './CategoriesLayout';
export { default as HeaderFavorites } from './HeaderFavorites';
export { default as AddressLayout } from './AddressLayout';
