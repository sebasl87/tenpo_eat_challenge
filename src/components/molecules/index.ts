export { default as DetailCard } from './DetailCard';
export { default as DiscountCircle } from './DiscountCircle';
export { default as LogoRounded } from './LogoRounded';
export { default as DetailFavorites } from './DetailFavorites';
