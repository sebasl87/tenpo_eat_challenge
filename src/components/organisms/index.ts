export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
export { default as ContainerMapView } from "./ContainerMapView";
export { default as RestaurantRow } from "./RestaurantRow";
export { default as ContainerAddressData } from "./ContainerAddressData";
export { default as CategoriesRow } from "./CategoriesRow";
export { default as FavoritesRow } from "./FavoritesRow";
export { default as HomeLocation } from './HomeLocation';
