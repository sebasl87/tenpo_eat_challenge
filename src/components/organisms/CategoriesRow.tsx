import React from 'react'
import { View, ScrollView, Text } from 'react-native';
import styled from 'styled-components/native';
import { CategoriesLayout } from '../atoms'

export interface ICategories { }

const Title = styled.Text`
  color: #333333;
  font-size: 18px;
  margin-bottom: 6px;
  margin-top: 32px;
`;

const linksData = [
    {
        uri: require("../../../assets/imgs/hamburguer.png"),
        category: "HAMBURGUESAS"
    },
    {
        uri: require("../../../assets/imgs/italian.png"),
        category: "ITALIANA"
    },
    {
        uri: require("../../../assets/imgs/pizza.png"),
        category: "PIZZAS"
    },

]

const CategoriesRow: React.FC<ICategories> = () => (
    <>

        <Title style={{ fontFamily: "Gotham-Bold" }}>CATEGORIAS</Title>
        <ScrollView horizontal={true}>
            <View style={{ flexDirection: "row" }}>
                {linksData.map((item, i) => (
                    <View key={i} style={{ marginRight: 6 }}>

                        <CategoriesLayout image={item.uri}>{item.category}</CategoriesLayout>
                    </View>
                ))}
            </View >
        </ScrollView>

    </>
);


export default CategoriesRow;