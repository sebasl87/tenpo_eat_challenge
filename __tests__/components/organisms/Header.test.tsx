import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import { Header } from "../../../src/components/organisms";
import { waitFor } from "@testing-library/react-native";
jest.useFakeTimers()

jest.mock('react-native', () => {
    const rn = jest.requireActual('react-native')
    const spy = jest.spyOn(rn.Animated, 'timing')
    spy.mockImplementation(() => {
        return {
            start: jest.fn(),
            timing: jest.fn(() => {
                return {
                    start: jest.fn(),
                };
            }),
            Value: jest.fn(() => {
                return {
                    interpolate: jest.fn(),
                };
            }),
        }
    });
    return rn
});
jest.useFakeTimers()

describe("Header component", () => {

    it("should render successfully ", async () => {

        const headerLocation = renderer
            .create(<Header />)
            .toJSON();
        await waitFor(() =>
            expect(headerLocation).toMatchSnapshot()
        )
    });
});