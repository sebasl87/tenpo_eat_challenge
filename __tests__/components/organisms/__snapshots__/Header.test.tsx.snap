// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render successfully  1`] = `
[
  <View
    style={
      {
        "backgroundColor": "#F2F2F2",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingTop": 40,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../assets/imgs/user.png",
        }
      }
      style={
        {
          "height": 40,
          "width": 40,
        }
      }
    />
    <Image
      source={
        {
          "testUri": "../../../assets/icons/search.png",
        }
      }
      style={
        {
          "height": 24,
          "width": 24,
        }
      }
    />
  </View>,
  <View
    style={
      {
        "backgroundColor": "#F2F2F2",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingTop": 40,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "column",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/imgs/logo.png",
          }
        }
        style={
          {
            "height": 88,
            "width": 118,
          }
        }
      />
      <Text
        style={
          {
            "fontFamily": "Gotham-Bold",
          }
        }
      >
        DELIVER APP
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": -16,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "left": -30,
            "opacity": 0,
            "position": "absolute",
            "top": -40,
            "transform": [
              {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icons/starCross.png",
            }
          }
          style={
            {
              "height": 12,
              "width": 12,
            }
          }
        />
      </View>
      <View
        collapsable={false}
        style={
          {
            "left": 10,
            "opacity": 0,
            "position": "absolute",
            "top": -70,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icons/starCross.png",
            }
          }
          style={
            {
              "height": 12,
              "width": 12,
            }
          }
        />
      </View>
      <Image
        source={
          {
            "testUri": "../../../assets/imgs/hand.png",
          }
        }
        style={
          {
            "height": 181,
            "top": -16,
            "width": 118,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "opacity": 0,
            "position": "absolute",
            "right": 20,
            "top": -40,
            "transform": [
              {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icons/starCross.png",
            }
          }
          style={
            {
              "height": 12,
              "width": 12,
            }
          }
        />
      </View>
      <View
        collapsable={false}
        style={
          {
            "opacity": 0,
            "position": "absolute",
            "right": 5,
            "top": 2,
            "transform": [
              {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icons/starCross.png",
            }
          }
          style={
            {
              "height": 12,
              "width": 12,
            }
          }
        />
      </View>
      <View
        collapsable={false}
        style={
          {
            "position": "absolute",
            "transform": [
              {
                "translateX": 0,
              },
            ],
            "zIndex": -1,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/imgs/moto.png",
            }
          }
          style={
            {
              "height": 130,
              "width": 106,
            }
          }
        />
      </View>
      <View
        style={
          {
            "height": 6,
            "top": "100%",
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/icons/ellipse.png",
            }
          }
          style={
            {
              "height": 6,
              "width": 28,
            }
          }
        />
      </View>
    </View>
  </View>,
]
`;
